src/main/java/org/dice/factcheckbenchmark/benchmark/rdf/TripleExtractor.java

```diff
@@ -0,0 +1,190 @@
+        else if (object.resource.isLiteral())
+            simplifiedData = simplifiedData + String.format("\"%s\" .\n", objectUri);
+
+        String labels = String.format("<%s> <http://www.w3.org/2000/01/rdf-schema#label> \"%s\"@en .\n" +
+                "<%s> <http://www.w3.org/2000/01/rdf-schema#label> \"%s\"@en .\n"
+                , subjectUri, subject.label
+                , objectUri, object.label);
+        simplifiedData = simplifiedData + labels;
+    }
+
+    public Model getSimplifiedModel() {
+        return this.simplifiedModel;
+    }
+
+    public String getSimplifiedData() {
+        return this.simplifiedData;
+    }
+
+    private void setSimplifiedModel() {
+        simplifiedModel = ModelFactory.createDefaultModel();
+        simplifiedModel.read(new ByteArrayInputStream(simplifiedData.getBytes()), null, "TTL");
+    }
+
+    private void parseStatements() {
+        boolean startNode = false;
+
+        StmtIterator stmtIterator = this.model.listStatements();
+        Resource subjectNode = null;
+        RDFNode objectNode = null;
+        while (stmtIterator.hasNext()) {
+            Statement statement = stmtIterator.next();
+
+            // look for starting node
+            if (statement.getSubject().getURI().matches("^.*__[0-9]*$")) {
+                startNode = true;
+                if (statement.getObject().isResource()) {
+                    subjectNode = statement.getSubject();
+                    if (subjectNode == null)
+                        continue;
+                    processTriple(statement, subjectNode, objectNode);
+                }
+            }
+        }
+
+        if (!startNode) {
+            stmtIterator = this.model.listStatements();
+            subjectNode = null;
+            objectNode = null;
+
+            while (stmtIterator.hasNext()) {
+                Statement statement = stmtIterator.next();
+
+                if (statement.getSubject() != null) {
+                    if (statement.getObject().isResource()) {
+                        subjectNode = statement.getSubject();
+                        if (subjectNode == null)
+                            continue;
+                        processTriple(statement, subjectNode, objectNode);
+                    }
+                }
+            }
+        }
+    }
+
+    private void processTriple(Statement statement, Resource subjectNode, RDFNode objectNode) {
+        this.predicate = statement.getPredicate();
+        objectNode = statement.getObject();
+
+        // check if object is resource and has edges, parse until you get Literal
+        getObject(statement, objectNode);
+
+        // now find if current statement subject node is part of object node
+        // then make that node as subject
+        getSubject(subjectNode);
+    }
+
+
+    /**
+     * retrieve subject node
+     * @param subjectNode subject initial node
+     */
+    private void getSubject(Resource subjectNode) {
```

Comment:
What exactly is this method doing? Are you really taking a resource and create a new resource with the same URI and the same Model instance? :open_mouth:

Reply:
My bad, function name should have been "setSubject" instead, as, at this point, we already have determined our subject.